<a href="https://colab.research.google.com/github/werowe/HypatiaAcademy/blob/master/class/order_of_precedence.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Order of Precedence

This means in a math equation like 3 * 3 / 2 which operator is executed first.  This chart shows that the * is executed before the division.  See the example below.

1. Parentheses ()
2. Exponents ^
3. Multiplication *
4. Division /
5. Addition +
6. Subtraction -


In [ ]:
# This one is trivial since one of the numerators 4 or 15 is a multiple of the denominator.  In other words is does
# not matter what we do first, multiplication or division

(4 * 15) / 5 = 4 * (15/ 5)

60 / 5 = 4 * 3

12 = 12



In [1]:
# so try this.  One student had this idea

4 * 5 + 1 / 5

# is this equal to

((4 * 5 ) + 1) / 5

#or

( 4 * (5 + 1)) / 5

# or


( 4 * 5 ) + ( 1 / 5)

20.2

The order of precedence says do multiplication first

so in:

4 * 5 + 1 / 5

we first get:

20 + 1 / 5

then it says do division so we get

20 + 1 / 5 = 20 + (1/5) = 20.2


In [3]:
 4 * 5 + 1 / 5

20.2

In [4]:
#exponents are easy as they are unary operations, meaning they work on themselves

4**2 + 1



17

In [5]:
# and subtraction and addition it does't matter.  why?  because the subtraction is just adding a negative number


2 - 3 == 2 + (-3)

True

In [6]:
# order doesn't matter either

4 + 3 - 2 == -2 + 3 + 4

True